product-mini/platforms/linux-sgx/enclave-sample/Enclave/Enclave.cpp

```diff
@@ -254,11 +334,14 @@ handle_cmd_load_module(uint64 *args, uint32 argc)
             wasm_runtime_free(enclave_module);
         else
             os_munmap(enclave_module, (uint32)total_size);
-        *(void **)args_org = NULL;
-        return;
+        ret = false;
+        goto exit;
     }
 
-    *(EnclaveModule **)args_org = enclave_module;
+    if (!gEnclaveModuleMgr.add(enclave_module, enclave_module_idx)) {
+        ret = false;
+        goto exit;
```

Comment:
Should destroy resources before exit, like:
```C
        wasm_runtime_unload(enclave_module->module);
        if (!enclave_module->is_xip_file)
            wasm_runtime_free(enclave_module);
        else
            os_munmap(enclave_module, (uint32)total_size);
```

Reply:
Oops, I didn't notice these. Have to check all problems like this